### PR TITLE
Implement client-side trigger

### PR DESCRIPTION
**Pull request by barryvdh**
- [x] Listen for triggers
 - [x] Emit client triggers back to other members of channel
 - [x] Check for private channel
 - [x] Check if client is authorized (!)
 - [x] Cross-check with laravel-echo PR (https://github.com/laravel/echo/pull/87)

### CODE REVIEW

**Comment by barryvdh** [comment]
Added verification to see if a client socket has joined the room, check that the event matches client-* and renamed the event to 'client event'

**Comment by barryvdh** [comment]
Conflicts are resolved :)

File: src/channels/channel.ts [hunk 1]
```diff
@@ -136,4 +164,32 @@ export class Channel {
...
+    /**
+     * Check if a socket has joined a channel.
+     *
+     * @param socket
+     * @param channel
+     * @returns {boolean}
+     */
+    isInChannel(socket: any, channel: string): boolean {
+        return this.io.sockets.adapter.sids[socket.id][channel];
+    }
```
Comment (barryvdh) [inline src/channels/channel.ts]: Perhaps we can do just `socket.rooms[channel]` here?

Comment (tlaverdure) [inline src/channels/channel.ts]: Have you been able to test that out? Socket.io code is extra complex and that was the first thing that I could find that proved to work.

Comment (barryvdh) [inline src/channels/channel.ts]: https://github.com/socketio/socket.io/blob/master/docs/API.md#socketrooms
`socket.rooms`: 
```js
{ Ny8TtuSKjpoSOKehAAAA: 'Ny8TtuSKjpoSOKehAAAA',
  'presence-chat.14': 'presence-chat.14',
  'private-App.Park.14': 'private-App.Park.14' }
```
`this.io.sockets.adapter.sids[socket.id]`: 
```js
{ Ny8TtuSKjpoSOKehAAAA: true,
  'presence-chat.14': true,
  'private-App.Park.14': true }
```

So I guess both should work.

Comment (barryvdh) [inline src/channels/channel.ts]: changed to `return !! socket.rooms[channel];`, which is working for me.

**Comment by barryvdh** [comment]
As fas as I'm concerned this is done, so if you're happy with this (eg. event names etc), please also reply on the laravel/echo issue so Taylor can merge the client-side :)

**Comment by tlaverdure** [comment]
Ok will do. Thanks!

**Pull request merged**